VerilogCodingStyle.md

```diff
@@ -2494,6 +2493,68 @@ always_ff @(posedge clk) begin
 end
 ```
 
+### Wildcard import of packages
+
+It is not permitted to use wildcard import syntax, eg. `import other_pkg::*;`
+in general to avoid naming collisions. However, wildcard import is allowed if
+the imported package is placed at the same IP and the import statement is
+placed at the module header or the module body.
+
+&#x1f44d;
+```systemverilog {.good}
+// mod_a_pkg.sv and mod_a.sv are placed at the same directory
+module mod_a import mod_a_pkg::*; (
+);
+
+endmodule
+```
+
+&#x1f44d;
+```systemverilog {.good}
+module mod_a ();
+
+  // mod_a_pkg.sv and mod_a.sv are placed at the same directory
+  import mod_a_pkg::*;
+
+endmodule
+```
+
+Other than the cases above, wildcard import is not allowed. For instance, below
+example may create name collision in the module following mod_a in the source
+list.
+
+&#x1f44e;
+```systemverilog {.bad}
+// mod_a.sv
+import mod_a_pkg::*; // not allowed: affecting all modules following
+
+module mod_a ();
+
+
+// mod_b.sv
+module mod_b ();
+  // mod_b cannot use the name defined in mod_a_pkg
+endmodule
```

Comment:
I know what you mean, but it's slightly confusing as this is effectively the reasoning, and not necessary for the example any more. I'd remove the `mod_b` part from this example.